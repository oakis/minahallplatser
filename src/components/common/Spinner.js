import React from 'react';
import {ActivityIndicator} from 'react-native';

export const Spinner = ({
  style,
  size,
  color,
  noFlex = false,
  left = false,
  right = false,
}) => {
  return (
    <ActivityIndicator
      size={size}
      color={color}
      style={[
        style,
        noFlex ? null : {flex: 1},
        left ? {alignSelf: 'flex-start'} : null,
        right ? {alignSelf: 'flex-end'} : null,
      ]}
    />
  );
};
