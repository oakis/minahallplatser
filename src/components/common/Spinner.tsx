import React from 'react';
import {ActivityIndicator} from 'react-native';

type SpinnerProps = {
  style?: Record<string, unknown>;
  size?: 'small' | 'large' | undefined;
  color?: string;
  noFlex?: boolean;
  left?: boolean;
  right?: boolean;
};

export const Spinner = ({
  style,
  size,
  color,
  noFlex = false,
  left = false,
  right = false,
}: SpinnerProps): JSX.Element => {
  return (
    <ActivityIndicator
      size={size}
      color={color}
      style={[
        style,
        noFlex ? null : {flex: 1},
        left ? {alignSelf: 'flex-start'} : null,
        right ? {alignSelf: 'flex-end'} : null,
      ]}
    />
  );
};
