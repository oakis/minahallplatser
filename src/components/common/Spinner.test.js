import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Spinner } from './Spinner';

it('should match snapshot with no flex', () => {
    const wrapper = shallow(<Spinner noFlex left />);
    expect(toJson(wrapper)).toMatchSnapshot();
});

it('should match snapshot with flex', () => {
    const wrapper = shallow(<Spinner />);
    expect(toJson(wrapper)).toMatchSnapshot();
});
