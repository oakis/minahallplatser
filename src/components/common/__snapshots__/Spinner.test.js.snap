// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot with flex 1`] = `
<ActivityIndicator
  style={
    Array [
      undefined,
      Object {
        "flex": 1,
      },
      null,
    ]
  }
/>
`;

exports[`should match snapshot with no flex 1`] = `
<ActivityIndicator
  style={
    Array [
      undefined,
      null,
      Object {
        "alignSelf": "flex-start",
      },
    ]
  }
/>
`;
